metadata/metadata.go

```diff
@@ -1331,7 +1421,13 @@ func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, ini
 	if has, err := serv.lookup.Has(id); err != nil {
 		return nil, err
 	} else if has {
-		return nil, &ResourceExists{id}
+		existing, err := serv.lookup.Lookup(id)
+		if err != nil {
+			return nil, err
+		}
+		if err := res.Update(serv.lookup, existing); err != nil {
+			return nil, err
+		}
 	}
 	if err := serv.lookup.Set(id, res); err != nil {
```

Comment:
You are setting it in the Update and here. Could remove the Set from the Update.